### PR TITLE
ci(github): Add TSC matcher and better workflow

### PR DESCRIPTION
**Pull request by billyvg**
Adds yarn caching, refactor node version to use nvmrc, adds tsc step

image

### CODE REVIEW

File: .github/workflows/javascript.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: javascript
+on:
+  pull_request:
+    paths:
+      - '**.[tj]sx?'
```
Comment (billyvg) [inline .github/workflows/javascript.yml]: Only run on PRs and when a js/ts file changes.

File: src/sentry/static/sentry/app/data/platforms.jsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint import/no-unresolved:0 import/order:0 */
```
Comment (billyvg) [inline src/sentry/static/sentry/app/data/platforms.jsx]: If we don't do this then we would need to install python deps to generate the integration docs.

File: tsconfig.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
     "noImplicitThis": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
-    "pretty": true,
+    "pretty": false,
```
Comment (billyvg) [inline tsconfig.json]: We need to disable pretty for the github action problem matcher to work.

File: .github/workflows/javascript.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        # See https://github.com/actions/cache/blob/master/examples.md#node---yarn for example
+        - name: Get yarn cache directory path
+          id: yarn-cache-dir-path
+          run: echo "::set-output name=dir::$(yarn cache dir)"
+
+        # Use nvmrc so we don't have yet another place node version is defined
+        - name: Get node version from .nvmrc
+          id: nvmrc
+          run: echo "::set-ouput name=version::$(cat .nvmrc)"
```
Comment (markstory) [inline .github/workflows/javascript.yml]:
```suggestion
          run: echo "::set-output name=version::$(cat .nvmrc)"
```

Comment (billyvg) [reply .github/workflows/javascript.yml]: Ooops, good catch, thanks!

**Review by markstory — APPROVED** [review]

**Pull request merged**